### PR TITLE
[bitnami/grafana-operator] Fix: volumes and secret mounts for grafana-operator

### PR DESCRIPTION
**Pull request by arnecls**
### Description of the change

The following fixes and features are contained in this PR

1. Indentation of `tolerations`
2. Indentation of `nodeSelector`
3. Indentation of `strategy`
4. Placement of `extraVolumes`, which are now part of the deployment spec
5. Placement of `extraVolumeMounts` which are now part of the deployment template spec
6. Re-introduction of `secrets` as volume mounts, so they can be used in sidecars
7. Allow to template `serviceAccount` section

### Benefits

Changes 1-5 fix rendering errors and make these settings work again

Change 6 (secrets) this is a convenience feature. You could also do this via `extraVolumes`, but keeping this setting allows some backwards compatibility with version 2 of this chart

Change 7 (serviceAccount templating) allows service account definitions with templating. This is a requirement when e.g. using workload identity binding and the GCP project name is defined as a global value.

### Possible drawbacks

Unknown

### Applicable issues

None

### Additional information

As a feature is re-introduced, the feature version of the chart has been increased.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/grafana-operator/templates/grafana.yaml [hunk 1]
```diff
@@ -96,6 +97,25 @@ spec:
...
+          {{- end }}
+          {{- if .Values.grafana.extraVolumes | or .Values.grafana.secrets }}
+          volumes:
+            {{- range .Values.grafana.secrets }}
+            - name: '{{.}}'
+              secret:
+                defaultMode: 420
+                optional: true
+                secretName: '{{.}}'
+            {{- end }}
```
Comment (dgomezleon) [inline bitnami/grafana-operator/templates/grafana.yaml:114]: What about using a similar approach as for example elasticsearch? There we expect this logic to be defined in the provided values.

```
## extraVolumes:
##   - name: es-certs
##     secret:
##       defaultMode: 420
##       secretName: es-certs
```

File: bitnami/grafana-operator/templates/grafana.yaml [hunk 2]
```diff
@@ -65,9 +65,7 @@ spec:
       {{- end }}
   {{- if .Values.grafana.serviceAccount }}
   serviceAccount:
-  {{- with .Values.grafana.serviceAccount }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.serviceAccount "context" $ ) | nindent 4 }}
```
Comment (arnecls) [inline bitnami/grafana-operator/templates/grafana.yaml:68]: @dgomezleon I just checked and the serviceAccount _is_ actually in the [grafana section].(https://github.com/bitnami/charts/blob/ca830ba72226e923335c49928810202a9129d440/bitnami/grafana-operator/values.yaml#L449)

```suggestion
    {{- include "common.tplvalues.render" ( dict "value" .Values.grafana.serviceAccount "context" $ ) | nindent 4 }}
```

Comment (dgomezleon) [reply bitnami/grafana-operator/templates/grafana.yaml:68]: You are right @arnecls I was checking [`grafana` values](https://github.com/bitnami/charts/blob/ca830ba72226e923335c49928810202a9129d440/bitnami/grafana/values.yaml#L568) instead.

**Review by dgomezleon — DISMISSED** [review]
The changes LGTM

**Review by dgomezleon — APPROVED** [review]

**Pull request merged**